providers/base/bin/watchdog_config_test.py

```diff
@@ -53,45 +118,243 @@ def watchdog_service_check():
         raise SystemExit("Error: {}".format(err))
 
 
-def main():
-    runtime_watchdog_usec = get_systemd_wdt_usec()
-    systemd_wdt_configured = runtime_watchdog_usec != "0"
-    wdt_service_configured = watchdog_service_check()
-    ubuntu_version = int(get_series().split(".")[0])
-    watchdog_config_ready = True
+def check_timeout() -> bool:
+    ubuntu_version: int = int(get_series().split(".")[0])
+    runtime_watchdog_usec: str = get_systemd_wdt_usec()
+    is_systemd_wdt_configured: bool = runtime_watchdog_usec != "0"
 
-    if (ubuntu_version >= 20) or (on_ubuntucore()):
-        if not systemd_wdt_configured:
-            print(
+    if ubuntu_version >= 20 or on_ubuntucore():
+        if not is_systemd_wdt_configured:
+            raise SystemExit(
                 "systemd watchdog should be enabled but reset timeout: "
                 "{}".format(runtime_watchdog_usec)
             )
-            watchdog_config_ready = False
-        if wdt_service_configured:
-            print("found unexpected active watchdog.service unit")
-            watchdog_config_ready = False
-        if watchdog_config_ready:
-            print(
-                "systemd watchdog enabled, reset timeout: {}".format(
-                    runtime_watchdog_usec
-                )
+        print(
+            "systemd watchdog enabled, reset timeout: {}".format(
+                runtime_watchdog_usec
             )
-            print("watchdog.service is not active")
+        )
     else:
-        if systemd_wdt_configured:
-            print(
-                "systemd watchdog should not be enabled but reset timeout: "
-                "{}".format(runtime_watchdog_usec)
+        if is_systemd_wdt_configured:
+            raise SystemExit(
+                "systemd watchdog should not be enabled but "
+                "reset timeout: {}".format(runtime_watchdog_usec)
+            )
+        print("systemd watchdog disabled")
+
+
+def check_service() -> bool:
+    ubuntu_version = int(get_series().split(".")[0])
+    is_wdt_service_configured = watchdog_service_check()
+
+    if ubuntu_version >= 20 or on_ubuntucore():
+        if is_wdt_service_configured:
+            raise SystemExit("Found unexpected active watchdog.service unit")
+        print("watchdog.service is not active")
+    else:
+        if not is_wdt_service_configured:
+            raise SystemExit("watchdog.service unit does not report as active")
+        print("watchdog.service is active")
+
+
+def detect() -> None:
```

Comment:
I iterated all watchdog devices under `/sys/class/watchdog`, so there should be no bug on this. And I think it is good to stay with checking the image, because I don't want people to accidentally use `udev_resource.py` when they forgot to set the environment variable.